### PR TITLE
Call out how write:packages requires repo scope on PATs

### LINKED ISSUE #2660: PAT creation section: possible incorrect info
### What article on docs.github.com is affected?

This one -> https://docs.github.com/en/free-pro-team@latest/packages/guides/migrating-to-github-container-registry-for-docker-images#authenticating-with-the-container-registry

### What part(s) of the article would you like to see updated?
[Authenticating with the container registry](https://docs.github.com/en/free-pro-team@latest/packages/guides/migrating-to-github-container-registry-for-docker-images#authenticating-with-the-container-registry)

The article mentions create a PAT with `write:packages` but without `repo` access

Screenshot 2021-01-08 at 12 08 19 AM

This is not possible right now👇
Screenshot 2021-01-08 at 12 09 32 AM

### PR DESCRIPTION
**Pull request by dgholz**
### Why:

**Closes #2660**

### What's being changed:

The docs used to recommend removing `repo` scope when creating a PAT with `write:packages`, but it's not possible. So, clearly call that out, and when linking to the security best practices, also call out which practice to follow.

### Check off the following:
- [ ] I have reviewed my changes in staging. (look for the **deploy-to-heroku** link in your pull request, then click **View deployment**)
- [x] For content changes, I have reviewed the [localization checklist](https://github.com/github/docs/blob/main/contributing/localization-checklist.md) (feels wordy)
- [x] For content changes, I have reviewed the [Content style guide for GitHub Docs](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

### CODE REVIEW

**Comment by dgholz** [comment]
I see in the [preview](https://docs-3746--patch-1.herokuapp.com/en/packages/guides/migrating-to-github-container-registry-for-docker-images#authenticating-with-the-container-registry) that the warning appears twice.

Screen Shot 2021-02-17 at 13 29 06

I don't know the best way to get it to appear only once when both reusables are included in the same page. Should I worry about the warning appearing when either reusable appears by itself? If not, which reusable should the warning appear in?

I feel like, if it appears in only one place, it should be in `reusables.package_registry.authenticate_with_pat_for_container_registry ` and not `data reusables.package_registry.authenticate-to-container-registry-steps` (matching where the original warning appeared)

**Comment by janiceilene** [comment]
@dgholz Thanks so much for opening a PR! I'll get this triaged for review ⚡

File: data/reusables/package_registry/authenticate-to-container-registry-steps.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
...
   {% warning %}
 
-  **Note:** If you select the `write:packages` scope, deselect the `repo` scope when creating the PAT. Adding a PAT with the `repo` scope as a secret in your repository allows the credential to be accessible to all collaborators in the repository. This gives unnecessary additional access when a PAT with the `repo` scope is used within an action. For more information on security best practices for actions, see "[Security hardening for GitHub Actions](/actions/getting-started-with-github-actions/security-hardening-for-github-actions#considering-cross-repository-access)."
+  **Note:** Granting the `write:packages` scope on the PAT also grants the `repo` scope! If you save this PAT for use from an Action, any collaborators can configure an Actions workflow to use this PAT to get full permissions to all repositories accessable from the PAT owner's account. We recommend creating a separate account with access only to the specific repositories that want to push Docker images (see "[Security hardening for GitHub Actions](/actions/getting-started-with-github-actions/security-hardening-for-github-actions#considering-cross-repository-access)").
```
Comment (jmarlena) [inline data/reusables/package_registry/authenticate-to-container-registry-steps.md:4]: Here's a suggestion that uses some slight rephrasing and follows our style guide. 

```suggestion
  **Warning:** Selecting the `write:packages` scope for your PAT will also automatically select the `repo` scope. If you use this PAT with an action, then a collaborator could potentially access your repositories. We recommend creating a separate account with access only to the specific repositories that want to push Docker images. For more information on security best practices for actions, see "[Security hardening for GitHub Actions](/actions/getting-started-with-github-actions/security-hardening-for-github-actions#considering-cross-repository-access)."
  ```

**Review by jmarlena — COMMENTED** [review]
Heads up, given the security related content I'm requesting some input from other stakeholders. ⚡

File: data/reusables/package_registry/authenticate-to-container-registry-steps.md (hunk 1 above)
Comment (jmarlena) [reply data/reusables/package_registry/authenticate-to-container-registry-steps.md:4]:
```suggestion
  **Note:** By default, when you select the `write:packages` scope for your personal access token (PAT) In the user interface, the `repo` scope will also be selected. The `repo` scope offers unnecessary and broad access, which we recommend you avoid using, especially for GitHub Actions workflows. For more information, see "[Security hardening for GitHub Actions](/actions/getting-started-with-github-actions/security-hardening-for-github-actions#considering-cross-repository-access)." As a workaround, you can select just the `write:packages` scope for your PAT in the user interface with this url: `https://github.com/settings/tokens/new?scopes=write:packages`. 
```

File: data/reusables/package_registry/authenticate_with_pat_for_container_registry.md [hunk 2]
```diff
@@ -1,7 +1,13 @@
...
 
-PATs can grant broad access to your account. We recommend selecting only the necessary read, write, or delete `package` scope when creating a PAT to authenticate to the {% data variables.product.prodname_container_registry %}. Avoid including the `repo` scope in a PAT used by a GitHub Actions workflow because it gives unnecessary additional access.
+PATs can grant broad access to your account. You should select only the necessary `read:packages`, `write:packages`, or `delete:packages` scope when creating a PAT to authenticate to the {% data variables.product.prodname_container_registry %}.
+
+{% warning %}
+
+**Note:** Granting the `write:packages` scope on the PAT also grants the `repo` scope! If you save this PAT for use from an Action, any collaborators can configure an Actions workflow to use this PAT to get full permissions to all repositories accessable from the PAT owner's account. We recommend creating a separate account with access only to the specific repositories that want to push Docker images (see "[Security hardening for GitHub Actions](/actions/getting-started-with-github-actions/security-hardening-for-github-actions#considering-cross-repository-access)").
```
Comment (jmarlena) [inline data/reusables/package_registry/authenticate_with_pat_for_container_registry.md:8]:
```suggestion
**Note:** * By default, when you select the `write:packages` scope for your personal access token (PAT) in the user interface, the `repo` scope will also be selected. The `repo` scope offers unnecessary and broad access, which we recommend you avoid using for GitHub Actions workflows in particular. 
 
If you save this PAT for use from an action, any collaborators can configure a GitHub Actions workflow to use this PAT to get full permissions to all repositories accessible from the PAT owner's account. For more information, see "[Security hardening for GitHub Actions](/actions/getting-started-with-github-actions/security-hardening-for-github-actions#considering-cross-repository-access)." 

As a workaround, you can select just the `write:packages` scope for your PAT in the user interface with this url: `https://github.com/settings/tokens/new?scopes=write:packages` and other desired scopes such as the `delete: packages` scope. Or you can create a separate account with access only to specific repositories that push Docker images.
```

**Comment by jmarlena** [comment]
**Note:** Since this PR was opened there have been some changes to this feature. GitHub Container Registry now supports using the `GITHUB_TOKEN` for GitHub Actions workflows, which improves the security experience here since users don't have to rely on a PAT in a workflow. 

I also noted a workaround for only selecting the `write:packages` scope with this url: `https://github.com/settings/tokens/new?scopes=write:packages`.

**Review by jmarlena — APPROVED** [review]
@dgholz Thank you for your contribution here!  🚢  This looks ready to ship!

**Review by Nawaz-Malik143 — COMMENTED** [review]
**[]()**

**Pull request merged**